### PR TITLE
Update CODEOWNERS with new SIG End User group

### PR DESCRIPTION
**Pull request by svrnm**
follow up to https://github.com/open-telemetry/community/issues/2004

### CODE REVIEW

**Review by theletterf — APPROVED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -22,7 +22,7 @@ content-modules/opentelemetry-proto      @open-telemetry/docs-maintainers @open-
 content-modules/opentelemetry-specification  @open-telemetry/docs-maintainers @open-telemetry/specs-approvers
 content-modules/semantic-conventions     @open-telemetry/docs-maintainers @open-telemetry/specs-semconv-approvers
 content/en/blog/                         @open-telemetry/docs-maintainers
-content/en/community/end-user/           @open-telemetry/docs-approvers @open-telemetry/end-user-wg
+content/en/community/end-user/           @open-telemetry/docs-approvers @open-telemetry/sig-end-user-approvers
```
Comment (danielgblanco) [inline .github/CODEOWNERS:25]: Should this be `sig-end-user-maintainers`? The end-user SIG only has maintainers at the moment.

Comment (svrnm) [reply .github/CODEOWNERS:25]: As far as I understand it maintainers are a subteam of the approvers, so if the approvers are tagged the maintainers will be tagged as well. We do this for consistency, but we can have the maintainers here if that's the preferred group

Comment (danielgblanco) [reply .github/CODEOWNERS:25]: Good point. I keep forgetting those are sub-teams!

**Review by danielgblanco — APPROVED** [review]
LGTM

File: .github/CODEOWNERS (hunk 1 above)
Comment (cartermp) [reply .github/CODEOWNERS:25]: GitHub doesn't seen to think that this team exists.

Comment (svrnm) [reply .github/CODEOWNERS:25]: ah, we need to fix repo permissions as well

**Pull request merged**